sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -444,8 +447,10 @@ final class DataFrameWriter private[sql](df: DataFrame) {
    * save mode, specified by the `mode` function (default to throwing an exception).
    * When `mode` is `Overwrite`, the schema of the [[DataFrame]] does not need to be
    * the same as that of the existing table.
-   * When `mode` is `Append`, the schema of the [[DataFrame]] need to be
-   * the same as that of the existing table, and format or options will be ignored.
+   *
+   * When `mode` is `Append`, the format or options will be ignored. The column order in the schema
```

Comment:
How about we explicitly say that if there is an existing table, we will use the format and options of the existing table.